lti_store/views.py

```diff
@@ -0,0 +1,192 @@
+import logging
```

Comment:
We introduced these views so each external configuration can have its own set of keyset and token URLs, if we used the access_token_endpoint and public_keyset_endpoint views on xblock-lti-consumer the tool would need to have a deployment for each XBlock if we were to use these views without having an endpoint for each XBlock the models will need to be refactored. Our purpose for using external configurations was to be able to only deploy a tool once and reuse that same configuration across multiple XBlocks.

I agree duplicating these views introduces a complexity that should not reside on the external app providing the external configuration, we are thinking about this as an initial implementation of external configurations for LTI 1.3, once the LtiConfiguration model on xblock-lti-consumer is refactored so each LtiConfiguration has a token and keyset endpoint per configuration and not per XBlock we could remove the use of these view on the plugin and use the views on xblock-lti-consumer.

Reply:
@kuipumu,
> if we used the access_token_endpoint and public_keyset_endpoint views on xblock-lti-consumer the tool would need to have a deployment for each XBlock

I'm afraid I don't understand this part. I'm probably missing something obvious, but why would we need multiple deployments? The way I understand the mechanism from https://github.com/openedx/xblock-lti-consumer/pull/390 is:
1. Each XBlock has its own `LtiConfiguration` entity in the DB.
2. Let's assume that this configuration specifies the `external_id`. Its `config_store` is set to `CONFIG_EXTERNAL`.
3. The external configuration stores the reusable configurations.
4. We call the [access_token_endpoint](https://github.com/openedx/xblock-lti-consumer/blob/1a9ca9ab41fd26b95c640a9f32a6413a36ef9c92/lti_consumer/plugin/views.py#L362-L418) built into the LTI Consumer XBlock.
5. This view [instantiates the `lti_consumer`](https://github.com/openedx/xblock-lti-consumer/blob/1a9ca9ab41fd26b95c640a9f32a6413a36ef9c92/lti_consumer/plugin/views.py#L396).
6. During the initialization, the `lti_consumer` [pulls data from the external configuration](https://github.com/openedx/xblock-lti-consumer/blob/1a9ca9ab41fd26b95c640a9f32a6413a36ef9c92/lti_consumer/models.py#L662-L679).
7. Then the `lti_consumer.access_token` call uses [this method](https://github.com/openedx/xblock-lti-consumer/blob/d82a697d1e9d891f16cbb94ab25b60ff135942a9/lti_consumer/lti_1p3/consumer.py#L425), which returns [this response](https://github.com/openedx/xblock-lti-consumer/blob/d82a697d1e9d891f16cbb94ab25b60ff135942a9/lti_consumer/lti_1p3/consumer.py#L487-L501). It seems to be identical to the one returned by [this custom `access_token` endpoint](https://github.com/open-craft/openedx-ltistore/blob/dc31a699047e7546bbb93cd24d0d8d690a2c7198/lti_store/views.py#L124-L140).

If we are concerned about using the external configuration ID in this endpoint, we could do something like this in the LTI Consumer XBlock:
```python
try:
    lti_config = LtiConfiguration.objects.get(config_id=lti_config_id)
except LtiConfiguration.DoesNotExist:
    lti_config = get_external_config_from_filter({}, lti_config_id)
```

Do we do something else in this new `access_token_endpoint` view?

Reply:
You can ignore that last comment, I wasn't able to use the already existing views to retrieve the keyset/token using the external ID, so I assumed that to be able to use these views we would need to refactor the `LtiConfiguration` model so we could have a URL per external configuration. 

The mechanism you describe is correct, except for how the `lti_consumer` would be initialized, the external configuration has no `access_token` method, so to use the access_token method with an external configuration, an instance of `LtiConsumer1p3` with the external configuration data will need to be created. I tested this and it works, so I eliminated these views from this PR and modified the views on the [xblock-lti-consumer PR](https://github.com/openedx/xblock-lti-consumer/pull/390) to allow the use of the already existing views.

When I started working on this feature I didn't find a way to have a keyset/token URL per external configuration so that's why I decided to move these views to the lti_store app. 

Apart from removing the keyset/token views from this plugin I also improved the fields used for LTI 1.3 on the `ExternalLtiConfiguration` model.

Thanks, and sorry for my delay in the response